git-edit-index

```diff
@@ -194,12 +194,14 @@ def edit_index(index):
     """
     # We need to use a temporary file to store the current index and show it to
     # the user in an editor.
-    with tempfile.NamedTemporaryFile(mode='r+', prefix='git-edit-index-') as f:
+    tmp = tempfile.mkstemp(prefix='git-edit-index-')
+    with os.fdopen(tmp[0], mode='w') as f:
         f.write(str(index))
-        f.flush()
-        f.seek(0)
-        subprocess.call(editor_cmd() + [f.name])
-        return Index.from_text(f.read(), line_sep='\n')
+    subprocess.call(editor_cmd() + [tmp[1]])
+    with open(tmp[1], mode='r') as f:
+        ret = Index.from_text(f.read(), line_sep='\n')
+    os.remove(tmp[1])
```

Comment:
If any code between `tempfile.mkstemp()` and `os.remove(tmp[1])` raises an exception, the file will not be deleted as the call to `os.remove()` will be skipped. This is not good as it may clutter the directory for temporary files. I would put the removal into a finally block:
```python
tmp = tempfile.mkstemp(prefix='git-edit-index-')
try:
    with os.fdopen(tmp[0], mode='w') as f:
        f.write(str(index))
    subprocess.call(editor_cmd() + [tmp[1]])
    with open(tmp[1], mode='r') as f:
        return Index.from_text(f.read(), line_sep='\n')
finally:
    os.remove(tmp[1])
```
As you can see, it even simplified the code as the `ret` temporary variable is no longer needed.